### PR TITLE
Adds pipeline serialization using json

### PR DESCRIPTION
**Pull request by graus**

### CODE REVIEW

**Comment by anneschuth** [comment]
Could you rebase your pickle changes away?

File: textpipe/pipeline.py [hunk 1]
```diff
@@ -37,3 +40,26 @@ def __call__(self, raw):
...
         result_dict = dict(zip(self.pipeline, [doc.__getattribute__(x) for x in self.pipeline]))
         return result_dict
+
+    def save(self, filename):
+        """
+        Serializes the pipeline to file (using pickle)
+
+        Args:
+        filename: location of where to serialize pipeline
+        """
```
Comment (anneschuth) [inline textpipe/pipeline.py]: Can you add a doctest with an example calls?

File: textpipe/pipeline.py [hunk 2]
```diff
@@ -37,3 +40,26 @@ def __call__(self, raw):
...
+            json.dump(self.__dict__, json_file)
+
+    @staticmethod
+    def load(filename):
+        """
+        Loads pipeline from serialized file
+
+        Args:
+        filename: location of pickled Pipeline object
+        """
```
Comment (anneschuth) [inline textpipe/pipeline.py]: And a doctest here

File: textpipe/pipeline.py [hunk 3]
```diff
@@ -13,7 +14,7 @@ class Pipeline:
     >>> sorted(pipe('Test sentence <a=>').items())
     [('clean_text', 'Test sentence '), ('nwords', 2), ('raw', 'Test sentence <a=>')]
     """
-    def __init__(self, pipeline, language=None, hint_language='en'):
+    def __init__(self, **kwargs):
```
[the remark this answers is not in the document]
Comment (graus) [reply textpipe/pipeline.py]: There is a better solution;

```def __init__(self, pipeline=None, **kwargs)```
[...]
```self.pipeline = pipeline or kwargs.get('pipeline', None)```

(we need optional `pipeline` for the `load()`-functionality

File: textpipe/pipeline.py [hunk 4]
```diff
@@ -37,3 +38,37 @@ def __call__(self, raw):
...
+        """
+        Serializes the pipeline to file (using json)
+
+        Args:
+        filename: location of where to serialize pipeline
+
+        >>> filename = "test.json"
+        >>> Pipeline(['n_sents', 'clean_text']).save(filename)
+        >>> open(filename).read()
+        '{"pipeline": ["n_sents", "clean_text"], "language": null, "hint_language": null}'
```
Comment (anneschuth) [inline textpipe/pipeline.py]: This might fail, the `self.__dict__` doesn't have a defined order.

File: textpipe/pipeline.py [hunk 5]
```diff
@@ -37,3 +38,37 @@ def __call__(self, raw):
...
+        Loads pipeline from serialized file
+
+        Args:
+        filename: location of serialized Pipeline object
+
+        >>> filename = "test.json"
+        >>> Pipeline(['n_sents', 'clean_text']).save(filename)
+        >>> p = Pipeline.load(filename)
+        >>> p.__dict__
+        {'pipeline': ['n_sents', 'clean_text'], 'language': None, 'hint_language': None}
```
Comment (anneschuth) [inline textpipe/pipeline.py]: I think dict equality doesn't really work. You can compare a sorted list of tuples instead

`sorted(p.__dict__.items())`

File: textpipe/pipeline.py (hunk 4 above)
Comment (dodijk) [reply textpipe/pipeline.py]: Solved that here like this: https://github.com/textpipe/textpipe/blob/master/textpipe/pipeline.py#L13.

**Pull request merged**